### PR TITLE
ci(lint): setup YAML Lint step

### PR DESCRIPTION
**Pull request by mmorel-35**
Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -4,7 +4,9 @@ updates:
     directory: "/"
     schedule:
       interval: "monthly"
-  - package-ecosystem: "gomod"  # Dependencies listed in go.mod
-    directory: "/" # Location of package manifests
```
Comment (jeffwidman) [inline .github/dependabot.yml:8]: does `yamllint` not like these? I don't recall seeing it complain before for inline comments, and not sure these comments are worth their own line as it adds noise...

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -29,17 +30,15 @@ jobs:
...
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: go mod tidy
+        run: |
+          go mod tidy
+          git diff --exit-code
+      - name: Test
+        run: go test -exec "sudo -n" -v ./...
```
Comment (jeffwidman) [inline .github/workflows/test.yml:44]: these formatting changes are all nice improvements 👍

**Review by jeffwidman — COMMENTED** [review]
Had one question... I personally tend to prefer some of these comments as inline, since they're distracting to folks who know what these params do, but helpful to provide more info for those who don't and need a little more hints.

Also, curious what prompted you to do this? It's nice, but we don't have _that_ much yaml in here... was your editor yelling at you about it?

**Comment by jeffwidman** [comment]
Can you squash your commits?

File: .github/dependabot.yml (hunk 1 above)
Comment (mmorel-35) [reply .github/dependabot.yml:8]: Yamllint need at least two whitespaces between content and comment.
I misunderstood it , now I can keep it on the same line

**Comment by mmorel-35** [comment]
> Can you squash your commits?

I'll do that when I can access a computer, within two days

**Comment by jeffwidman** [comment]
No worries, I can just squash when merging.

Also, in the future, if you don't have a computer where you can locally `git clone` a repo to, you can also just use a codespace: https://github.com/features/codespaces

I only recently discovered them myself, and while I still prefer more advanced coding on my local laptop, for quick changes, esp git commit changes, the codespaces are incredibly convenient.

**Review by jeffwidman — APPROVED** [review]
Thanks again for this!

**Pull request merged**